From 3bdcd9b24bc738fd5cc717473635bbb3f952a19f Mon Sep 17 00:00:00 2001
From: Varun Kumar Reddy Yaparla <vyaparla@habana.ai>
Date: Mon, 17 Jan 2022 14:08:16 +0200
Subject: [PATCH] [SW-70899] gaudi2: arc: reset CPU_PLL

This change resets CPU_PLL as part of prstn reset.
Reset value of PLL_L_PRSTN_RST_CFG is 0xFFFFFFFD.
Bit 1 which corresponds to CPU_PLL should also need
to be resetted since now ARCs are also getting resetted
as part of Warm/PRSTn reset
Without this fix, protection bits configuration is not
getting reset even when fake security is disabled followed
by prstn reset which results in 0xBABA0BAD while accessing
registers in CPU_PLL block

Change-Id: I6b1493b56ea9f1ff1aa5dcbb9de92e8d8db09692
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index e91149a..c5bb6c6 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -65,6 +65,13 @@
 	regmap_writel(0x1, mmPSOC_RESET_CONF_PSOC_PRSTN_RST_CFG);
 	regmap_writel(0x3, mmPSOC_RESET_CONF_ARC_PRSTN_RST_CFG);
 
+	/*
+	 * Reset value of PLL_L_PRSTN_RST_CFG is 0xFFFFFFFD, BIT(1) which corresponds to CPU_PLL
+	 * is not getting reset by default. Now ARCs are getting reset as part of Warm/PRSTn
+	 * reset, CPU_PLL should also need to be resetted
+	 */
+	regmap_writel(0xFFFFFFFF, mmPSOC_RESET_CONF_PLL_L_PRSTN_RST_CFG);
+
 	/* Clear masking of PRSTN from SW reset */
 	regmap_writel(0x0, mmPSOC_GLOBAL_CONF_PRSTN_MASK);
 
